controllers/yawol-cloud-controller/targetcontroller/service_controller.go

```diff
@@ -223,19 +223,26 @@ func (r *ServiceReconciler) createLoadBalancer(
 func (r *ServiceReconciler) reconcileInfrastructure(
 	ctx context.Context,
 	lb *yawolv1beta1.LoadBalancer,
+	svc *coreV1.Service,
 	infraConfig InfrastructureDefaults,
 ) error {
 	newInfra := yawolv1beta1.LoadBalancerInfrastructure{
-		FloatingNetID:    infraConfig.FloatingNetworkID,
-		NetworkID:        *infraConfig.NetworkID,
-		Flavor:           infraConfig.FlavorRef,
-		Image:            infraConfig.ImageRef,
-		AvailabilityZone: *infraConfig.AvailabilityZone,
+		DefaultNetwork:     getDefaultNetwork(svc, infraConfig),
+		AdditionalNetworks: getAdditionalNetworks(svc, infraConfig),
+		Flavor:             infraConfig.FlavorRef,
+		Image:              infraConfig.ImageRef,
+		AvailabilityZone:   *infraConfig.AvailabilityZone,
+		ProjectID:          getProjectID(svc),
 		AuthSecretRef: coreV1.SecretReference{
 			Name:      *infraConfig.AuthSecretName,
 			Namespace: *infraConfig.Namespace,
 		},
 	}
+
+	if !reflect.DeepEqual(newInfra.ProjectID, lb.Spec.Infrastructure.ProjectID) {
+		return kubernetes.SendErrorAsEvent(r.Recorder, helper.ErrProjectIsImmutable, svc)
```

Comment:
This should be fine, because the controller uses his backoff limits. Also the event will be always present, so the user can react to it (events often do not last that long in k8s)